Python/sysmodule.c

```diff
@@ -2861,11 +2862,9 @@ PyAPI_FUNC(void)
 PySys_AddWarnOptionUnicode(PyObject *option)
 {
     PyThreadState *tstate = _PyThreadState_GET();
-    if (_PySys_AddWarnOptionWithError(tstate, option) < 0) {
+    if (tstate && _PySys_AddWarnOptionWithError(tstate, option) < 0) {
         /* No return value, therefore clear error state if possible */
-        if (tstate) {
-            _PyErr_Clear(tstate);
-        }
+        _PyErr_Clear(tstate);
```

Comment:
I assume that there's no way to make `tstate` NULL in `_PySys_AddWarnOptionWithError`.

Reply:
Would it actually make sense to have `_PyErr_Clear` do this automatically?

Reply:
Ok, in that case, I would like `PySys_AddWarnOptionUnicode` to fail with a fatal error if called without the GIL. No-oping because of a null thread state isn't common at all in the C API.

Reply:
Should I do some other changes or leave as is?

Reply:
Could we actually do it in `_PySys_AddWarnOptionWithError` instead? I think it makes more sense (and avoids duplicating the assertions). WDYT @ZeroIntensity?

Reply:
I think `assert(tstate != NULL)` is fine for the private API.